pyk4a/pyk4a.cpp

```diff
@@ -596,6 +600,80 @@ extern "C" {
+        k4a_transformation_t* transformation_handle;
+        PyObject *capsule;
+        int thread_safe;
+        PyThreadState *thread_state;
+        k4a_result_t res;
+        PyArrayObject *d_array;
+        PyArrayObject *c_array;
+        k4a_color_resolution_t color_resolution;
+	k4a_transformation_interpolation_type_t interpolation_type = K4A_TRANSFORMATION_INTERPOLATION_TYPE_NEAREST;
+
+
+        PyArg_ParseTuple(args, "OpO!O!I", &capsule, &thread_safe, &PyArray_Type, &d_array, &PyArray_Type, &c_array, &color_resolution);
+        transformation_handle = (k4a_transformation_t*)PyCapsule_GetPointer(capsule, CAPSULE_TRANSFORMATION_NAME);
+
+        k4a_image_t* depth_image_transformed = (k4a_image_t*) malloc(sizeof(k4a_image_t));
+        k4a_image_t* custom_image_transformed = (k4a_image_t*) malloc(sizeof(k4a_image_t));
+
+        k4a_image_t depth_image;
+        k4a_image_t custom_image;
+        res = numpy_to_k4a_image(d_array, &depth_image, K4A_IMAGE_FORMAT_DEPTH16);
+	if (K4A_RESULT_SUCCEEDED == res) {
+            res = numpy_to_k4a_image(c_array, &custom_image, K4A_IMAGE_FORMAT_CUSTOM16);
+	}
+        thread_state = _gil_release(thread_safe);
+        if (K4A_RESULT_SUCCEEDED == res) {
+            res = k4a_image_create(
+                    k4a_image_get_format(depth_image),
+                    RESOLUTION_TO_DIMS[color_resolution][0],
+                    RESOLUTION_TO_DIMS[color_resolution][1],
+                    RESOLUTION_TO_DIMS[color_resolution][0] * (int)sizeof(uint16_t),
+                    depth_image_transformed);
+        }
+        if (K4A_RESULT_SUCCEEDED == res) {
+            res = k4a_image_create(
+                    k4a_image_get_format(custom_image),
+                    RESOLUTION_TO_DIMS[color_resolution][0],
+                    RESOLUTION_TO_DIMS[color_resolution][1],
+                    RESOLUTION_TO_DIMS[color_resolution][0] * static_cast<int32_t>(sizeof(int16_t)),
+                    custom_image_transformed);
+        }
+
+        if (K4A_RESULT_SUCCEEDED == res) {
+            res = k4a_transformation_depth_image_to_color_camera_custom(
+                    *transformation_handle,
+                    depth_image,
+                    custom_image,
+		    *depth_image_transformed,
+		    *custom_image_transformed,
+		    interpolation_type,
+		    0);
+            k4a_image_release(depth_image);
+            k4a_image_release(custom_image);
+        }
+        _gil_restore(thread_state);
+        PyArrayObject* np_depth_image;
+        PyArrayObject* np_custom_image;
+        if (K4A_RESULT_SUCCEEDED == res) {
+            res = k4a_image_to_numpy(depth_image_transformed, &np_depth_image);
+        }
+        if (K4A_RESULT_SUCCEEDED == res) {
+            res = k4a_image_to_numpy(custom_image_transformed, &np_custom_image);
+        }
+
+        if (K4A_RESULT_SUCCEEDED == res) {
+            return PyArray_Return(np_custom_image);
```

Comment:
Would you mind trying something like
`return Py_BuildValue("OO", np_transformed_custom_image, np_transformed_depth_image);`